softioc/device_core.py

```diff
@@ -83,20 +48,16 @@ class DeviceSupportCore(DeviceCommon):
     # treatment) are then bound to the appropriate class instance and the
     # appropriate method is invoked.
 
-    def __init_class__(cls):
+    def __init_subclass__(cls):
         '''Record support initialisation, called once during class
         initialisation for each sub-class.  This registers record support for
         the specified device name.'''
+
         if not hasattr(cls, '_record_type_'):
             # If no record type has been specified then we're a base class
             # with nothing to do.
             return
 
-        # Create an empty device directory.
-        # (Potentially this belongs in a mix-in for resolving the linkage
-        # between record and instances, but for now this is hard-wired here.)
-        cls.__device_directory = {}
-
```

Comment:
Interesting: I can find no trace of this symbol anywhere else (even using `git log -S`)!  I guess this was an idea that either never happened or has been erased.